### PR TITLE
Remove architecture diagram from the resources directory

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6832

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [x] Checked native-image compatibility

### CODE REVIEW

File: ballerina/docs/architecture.svg
Comment (TharmiganK) [inline ballerina/docs/architecture.svg:1]: Aren't we linking this in the API docs?

Comment (daneshk) [reply ballerina/docs/architecture.svg:1]: I thought of doing it later since we are doing a patch release. let me add it as well

Comment (daneshk) [reply ballerina/docs/architecture.svg:1]: Fixed [30e99bb](https://github.com/ballerina-platform/module-ballerina-io/pull/1223/commits/30e99bb84eaaca27f846044d8c7c338779292d52)

File: ballerina/Module.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This module provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.
 
+The following diagram depicts the overview architecture of the I/O package.
+
+![Overview](docs/architecture.svg)
```
Comment (TharmiganK) [inline ballerina/Module.md:7]: Does this path work? In the bala, this file is located inside the `docs/<module-dir-path>` directory and the diagram is located in the `docs` directory.
I am not sure whether `bal doc` command will give us some idea here. May be we have to push it to staging and check

Can't we just give the github url for this image: https://raw.githubusercontent.com/ballerina-platform/module-ballerina-io/master/ballerina/resources/architecture.svg
I think in connectors we use the github links

File: ballerina/Ballerina.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ repository = "https://github.com/ballerina-platform/module-ballerina-io"
 icon = "icon.png"
 license = ["Apache-2.0"]
 distribution = "2201.8.0"
+include = ["docs"]
```
[the remark this answers is not in the document]
Comment (daneshk) [reply ballerina/Ballerina.toml:11]: Fixed in [9e2fff5](https://github.com/ballerina-platform/module-ballerina-io/pull/1223/commits/9e2fff5be349090ac0836011d70ac7a400dea3f4)

File: ballerina/Module.md (hunk 1 above)
Comment (daneshk) [reply ballerina/Module.md:7]: yes, I am planning to push this to the staging central and check whether it is working. Inside the bala file, it is inside the module,
Screenshot 2024-08-06 at 18 15 46

Comment (TharmiganK) [reply ballerina/Module.md:7]: Isn't it better to give the URL?

Comment (daneshk) [reply ballerina/Module.md]: Added in [e938e18](https://github.com/ballerina-platform/module-ballerina-io/pull/1223/commits/e938e18e1c8582fea6d853e23c7406d57107b16c)

File: ballerina/Module.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 This module provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.
 
+The following diagram depicts the overview architecture of the I/O package.
+
+![Overview](https://raw.githubusercontent.com/ballerina-platform/module-ballerina-io/master/ballerina/docs/architecture.svg)
```
Comment (daneshk) [inline ballerina/Module.md:7]:
```suggestion
![Architecture Diagram](https://raw.githubusercontent.com/ballerina-platform/module-ballerina-io/master/ballerina/docs/architecture.svg)
```

File: ballerina/Package.md [hunk 4]
```diff
@@ -2,6 +2,10 @@
 
 This package provides file read/write APIs and console print/read APIs. The file APIs allow read and write operations on different kinds of file types such as bytes, text, CSV, JSON, and XML. Further, these file APIs can be categorized as streaming and non-streaming APIs.
 
+The following diagram depicts the overview architecture of the I/O package.
+
+![Overview](https://raw.githubusercontent.com/ballerina-platform/module-ballerina-io/master/ballerina/docs/architecture.svg)
```
Comment (daneshk) [inline ballerina/Package.md:7]:
```suggestion
![Architecture Diagram](https://raw.githubusercontent.com/ballerina-platform/module-ballerina-io/master/ballerina/docs/architecture.svg)
```

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**